### PR TITLE
put store ignore tombstone.

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: server/cluster.go [hunk 1]
```diff
@@ -407,6 +407,10 @@ func (c *RaftCluster) putStore(store *metapb.Store) error {
 
 	// Case 3: store id does not exist, check duplicated address.
 	for _, s := range c.cachedCluster.getStores() {
+		// It's OK to start a new store on the same address if the old store has been removed.
+		if s.store.GetState() == metapb.StoreState_Tombstone {
```
Comment (qiuyesuifeng) [inline server/cluster.go]: If the store is tombstone, should we check more conditions before restart this store?

Comment (huachaohuang) [inline server/cluster.go]: We do not restart the tombstone store here, but we put a new store with the same address with the tombstone store.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**